unconstrained_test.go

```diff
@@ -11,1267 +11,397 @@ import (
-		gradTol: defaultGradientAbsTol,
+		f: funcs.BiggsEXP2{},
+		x: []float64{1, 2},
 	},
 	{
-		f:      RosenbrockFDf{},
-		x:      []float64{-1.2, 1},
-		optVal: 0,
-		optLoc: []float64{1, 1},
+		f: funcs.BiggsEXP2{},
+		x: []float64{1.00001, 10.00001},
 	},
 	{
-		f:       RosenbrockFDf{},
-		x:       []float64{-120, 100, 50},
-		optVal:  0,
-		optLoc:  []float64{1, 1, 1},
-		gradTol: defaultGradientAbsTol,
+		f: funcs.BiggsEXP3{},
+		x: []float64{1, 2, 1},
 	},
 	{
-		f:      HelicalValley{},
-		x:      []float64{-1, 0, 0},
-		optVal: 0,
-		optLoc: []float64{1, 0, 0},
+		f: funcs.BiggsEXP3{},
+		x: []float64{1.00001, 10.00001, 3.00001},
 	},
 	{
-		f:      BiggsEXP4{},
-		x:      []float64{1, 2, 1, 1},
-		optVal: 0,
-		optLoc: []float64{1, 10, 1, 5},
+		f: funcs.ExtendedRosenbrock{},
+		x: []float64{-1.2, 1},
 	},
-}
-
-var cgTests = []UnconstrainedTest{
 	{
-		f:      Rosenbrock{},
-		x:      []float64{-1200000, 1000000},
-		optVal: 0,
-		optLoc: []float64{1, 1},
+		f: funcs.ExtendedRosenbrock{},
+		x: []float64{1.00001, 1.00001},
 	},
 	{
-		f:       BiggsEXP6{},
-		x:       []float64{1, 2, 1, 1, 1, 1},
-		optVal:  0.005655649925,
-		gradTol: 1e-6,
+		f: funcs.ExtendedRosenbrock{},
+		x: []float64{-1.2, 1, -1.2},
+	},
+	{
+		f:       funcs.ExtendedRosenbrock{},
+		x:       []float64{1.00001, 1.00001, 1.00001},
+		gradTol: 1e-8,
 	},
 	{
-		f:       Gaussian{},
+		f:       funcs.Gaussian{},
 		x:       []float64{0.4, 1, 0},
-		optVal:  1.12793e-8,
-		optLoc:  []float64{0.3989561, 1.0000191, 0},
 		gradTol: 1e-9,
 	},
-	generateVariablyDimensioned(1000, 1e-9),
-	generateVariablyDimensioned(10000, 1e-8),
 	{
-		f:       Watson{},
-		x:       []float64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
-		optVal:  0,
-		optLoc:  []float64{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1},
-		gradTol: 1e-6,
+		f:       funcs.Gaussian{},
+		x:       []float64{0.398, 1, 0},
+		gradTol: 1e-9,
 	},
 	{
-		f:       Penalty1{},
-		x:       []float64{1, 2, 3, 4},
-		optVal:  2.2499775e-5,
-		gradTol: 1e-10,
+		f: funcs.HelicalValley{},
+		x: []float64{-1, 0, 0},
 	},
 	{
-		f:       Penalty1{},
-		x:       []float64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
-		optVal:  7.0876515e-5,
+		f: funcs.HelicalValley{},
+		x: []float64{1.00001, 0.00001, 0.00001},
+	},
+	{
+		f:       funcs.Trigonometric{},
+		x:       []float64{0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1},
 		gradTol: 1e-8,
 	},
+	newVariablyDimensioned(2, 0),
 	{
-		f:       Penalty2{},
-		x:       []float64{0.5, 0.5, 0.5, 0.5},
-		optVal:  9.37629e-6,
-		gradTol: 1e-6,
+		f: funcs.VariablyDimensioned{},
+		x: []float64{1.00001, 1.00001},
 	},
+	newVariablyDimensioned(10, 0),
 	{
-		f:       Penalty2{},
-		x:       []float64{0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5},
-		optVal:  2.93660e-4,
-		gradTol: 1e-6,
+		f: funcs.VariablyDimensioned{},
+		x: []float64{1.00001, 1.00001, 1.00001, 1.00001, 1.00001, 1.00001, 1.00001, 1.00001, 1.00001, 1.00001},
 	},
+}
+
+var cgTests = []unconstrainedTest{
```

Comment:
ExtendedPowell has been renamed to ExtendedPowellSingular, which is included in cgTests.
Beale is included in gradientDescentTests (before it wasn't) which are tested always, cgTests are appended to them.

There were indeed other functions that I forgot to include, now they should be back.

On a side note, CG is quite troublesome to test, there are many combinations of CGVariant + StepSizer and for many tests there is always one combination that always fails. Furthermore, if the tests pass locally does not mean that they pass on Travis and the cycle modify-test_locally-commit-push-wait_for_travis-check_if_travis_failed becomes annoying rather quickly.